prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt

```diff
@@ -0,0 +1,91 @@
+package gov.cdc.prime.router.fhirengine.translation
+
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaReader
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.converter.ConverterSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FhirTransformSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.ConstantSubstitutor
+import gov.cdc.prime.router.fhirengine.utils.FhirTranscoder
+import javax.ws.rs.InternalServerErrorException
+
+class TranslationSchemaManager {
+    enum class SchemaType(val directory: String) {
+        FHIR("fhir_transforms"),
+        HL7("hl7_mapping"),
+    }
+
+    /**
+     * Downloads the content of each blob in the [directory] with [blobContainerInfo] takes the input file, runs
+     * the transform on it, and compares it to the output validating that they are the same.
+     */
+    fun validateManagedSchemas(
+        schemaType: SchemaType,
+        blobContainerInfo: BlobAccess.BlobContainerMetadata = BlobAccess.defaultBlobMetadata,
+        blobEndpoint: String,
+    ): Boolean {
+        val sourceContainer = BlobAccess.getBlobContainer(blobContainerInfo)
+        val blobs =
+            BlobAccess.listBlobs(schemaType.directory, blobContainerInfo, false)
+        val inputs = blobs.filter { it.currentBlobItem.name.startsWith("input.") }
+
+        inputs.forEach { currentInput ->
+            val inputDirectoryPath = Regex("(.*/).*").find(currentInput.currentBlobItem.name)!!.groups[1]!!.value
+            val input = sourceContainer.getBlobClient(inputDirectoryPath + "input.fhir")
+            val inputBundle = FhirTranscoder.decode(input.toString())
+
+            val transformBlob = blobs.filter {
+                it.currentBlobItem.name.startsWith(inputDirectoryPath) &&
+                it.currentBlobItem.name.endsWith(".yml")
+            }
+            val transformBlobName = transformBlob[0].currentBlobItem.name
+
+            when (schemaType) {
```

Comment:
No default case in this switch? In the case where the schemaType is something we don't support, we should figure out how to communicate that. I'm leaning towards updating the proposed data object to record the schemaType was not supported.

Reply:
The when is exhaustive so there isn't a default case needed.